link/perf_event.go

```diff
@@ -251,3 +282,32 @@ func uint64FromFile(base string, path ...string) (uint64, error) {
 	et := bytes.TrimSpace(data)
 	return strconv.ParseUint(string(et), 10, 64)
 }
+
+// randomGroup generates a pseudorandom string for use as a tracefs group name.
+// Returns an error when the output string would exceed 63 characters (kernel
+// limitation), when rand.Read() fails or when prefix contains characters not
+// allowed by rgxTraceEvent.
+func randomGroup(prefix string) (string, error) {
```

Comment:
Could this be kept in `kprobe.go`? It's not perf event-related and inflates the diff.

Reply:
the thing is, it's also related to uprobe but I also don't like making a new file for methods in common between uprobe and kprobe. Should I move all the methods in common in `kprobe.go`?

Reply:
Yup, can all stay in `kprobe.go`. A uprobe is still technically a kprobe. :sweat_smile:

Reply:
👌🏻